File: acm/accepted/10324.pas
program p10324;
{
	@JUDGE_ID:  20044WC 10324 PASCAL "DaP/DyP"
}
var
	i,j,k : longint;
	n : longint;
	c,l : char;
	dp : array[1..1000000] of longint;
	a,b : longint;
	caseno: longint;
begin
	caseno := 0;
	while not eof do begin
		if eoln then break;
		inc (caseno);
		n := 1;
		read(l);
		dp[n] := 1;
		while not eoln do begin
			inc (n);
			read(c);
			if c = l then begin
				dp[n] := dp[n-1] + 1;
			end else begin
				l := c;
				dp[n] := 1;
			end;
		end;
		readln;
		writeln('Case ', caseno,':');
		{for i := 1 to n do begin
			write(dp[i]:4);
		end;writeln;}
		readln(k);
		for i := 1 to k do begin
			readln(a,b);
			if a > b then begin
				j := a;
				a := b;
				b := j;
			end;
			inc(a);
			inc(b);
			if (dp[b]-(b-a+1) ) >= 0 then writeln('Yes') else writeln('No');
		end;
	end;
end.

File: acm/accepted/c/10324.c
#include <stdio.h>
/*
@JUDGE_ID:  20044WC 10324 C "I WANT I WANT"
*/
int main() {
	int i;
	int dp[1100000];
	int n;
	int s;
	char c, l;
	int a,b,t;
	int whichset;

	whichset = 0;
	while (! feof(stdin)) {
		whichset ++;
		l = ' ';
		s = 0; n = 0; a = b = t = 0;
		while (1) {
			c = getc(stdin);
			if (c == EOF) { return 0; };
			if (c == '\n') { break; };
			if ((c == l) && (s > 0)) { dp[s] = dp[s-1] + 1; } else { dp [s] = 0; };
			s ++;
			l = c;
		};
		
		printf("Case %d:\n", whichset);
		scanf("%d", &n);

		for (i=1;i<=n;++i) {
			scanf("%d %d\n", &a, &b);
			if (a > b) { t = b; b = a; a = t; };
			if ((dp[b] - (b-a) ) >= 0) { printf("Yes\n"); } else { printf("No\n"); };
		};
	};
	return 0;
};
